src/plugins/intel_cpu/tests/unit/shape_inference_test/col2im_shape_inference_test.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include "common_test_utils/test_assertions.hpp"
+#include "utils.hpp"
+
+using namespace ov;
+using namespace ov::intel_cpu;
+using ov::op::v0::Constant;
+using ov::op::v0::Parameter;
+using testing::HasSubstr;
+
+class Col2ImStaticTestSuite : public ::testing::TestWithParam<std::tuple<Shape,         // data shape
+                                                              std::vector<int32_t>,     // output_size values
+                                                              std::vector<int32_t>,     // kernel_size values
+                                                              Strides,                  // strides
+                                                              Strides,                  // dilations
+                                                              Shape,                    // pads_begin
+                                                              Shape,                    // pads_end
+                                                              Shape>> {};               // expected output shape
+
+TEST_P(Col2ImStaticTestSuite, Col2ImStaticShapeInference) {
+    const auto& param = GetParam();
+    const auto& data_shape = std::get<0>(param);
+    const auto& output_size_val = std::get<1>(param);
+    const auto& kernel_size_val = std::get<2>(param);
+    const auto& strides = std::get<3>(param);
+    const auto& dilations = std::get<4>(param);
+    const auto& pads_begin = std::get<5>(param);
+    const auto& pads_end = std::get<6>(param);
+    const auto& expected_output_shape = std::get<7>(param);
+
+    const auto data = std::make_shared<Parameter>(element::i64, data_shape);
+    const auto output_size = std::make_shared<op::v0::Constant>(element::i64, Shape{2}, output_size_val);
+    const auto kernel_size = std::make_shared<op::v0::Constant>(element::i64, Shape{2}, kernel_size_val);
```

Comment:
> For this tests the all inputs can have dynamic shape if possible.

I avoided that since first input's shape is used for calculating second Dimension of the output shape.

> Add scenarios when output_size and/or kernel_size are not constants but data are get from tensor accessor.

Done.

> Add some error simple error scenarios wrong data or missing data that static shape calculation will be not possible.

Such tests are already in type_prop tests, should I copy/move them here?